zutil.h

```diff
@@ -20,6 +20,48 @@
 #  define ZLIB_INTERNAL
 #endif
 
+#if defined(HAVE_ATTRIBUTE_SSE2)
+#  define ZNG_TARGET_SSE2 __attribute__((__target__("sse2")))
+#else
+#  define ZNG_TARGET_SSE2
+#endif
+#if defined(HAVE_ATTRIBUTE_SSSE3)
+#  define ZNG_TARGET_SSSE3 __attribute__((__target__("ssse3")))
+#else
+#  define ZNG_TARGET_SSSE3
+#endif
+#if defined(HAVE_ATTRIBUTE_SSE42)
+#  define ZNG_TARGET_SSE42 __attribute__((__target__("sse4.2")))
+#else
+#  define ZNG_TARGET_SSE42
+#endif
+#if defined(HAVE_ATTRIBUTE_PCLMULQDQ)
+#  define ZNG_TARGET_PCLMULQDQ __attribute__((__target__("pclmul")))
+#  if defined(HAVE_ATTRIBUTE_SSE2)
+#    if defined(HAVE_ATTRIBUTE_SSE42)
+#       define ZNG_TARGET_SSE2_SSE42_PCLMULQDQ \
+          __attribute__((__target__("sse2,sse4.2,pclmul")))
+#    endif
+#    if defined(HAVE_ATTRIBUTE_SSSE3)
+#      define ZNG_TARGET_SSE2_SSSE3_PCLMULQDQ \
+         __attribute__((__target__("sse2,ssse3,pclmul")))
```

Comment:
sse2 can be removed from this and the above attribute definitions since it is a subset of both ssse3 and sse4.2.
pclmul is IIRC not a superset of either, so it needs to stay.
However, investigating whether pclmul is only supported in cpus with for example sse4.2 for example would mean that we would get more efficient code by also requiring sse4.2 instead of ssse3, by allowing the compiler the use of more recent instruction sets in code that only gets run on cpus that supports it anyway.

Reply:
Ok I have made that change.